dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/QueryFilter.java

```diff
@@ -113,21 +113,44 @@ public boolean isOperator( QueryOperator op )
     }
 
     public String getSqlOperator()
+    {
+        return getSqlOperator( false );
+    }
+
+    /**
+     *
+     * @param isOperatorSubstitutionAllowed whether the operator should be
+     *        replaced to support null values or not
+     * @return
+     */
+    public String getSqlOperator( boolean isOperatorSubstitutionAllowed )
     {
         if ( operator == null )
         {
             return null;
         }
 
-        return safelyGetOperator();
+        return safelyGetOperator( isOperatorSubstitutionAllowed );
     }
 
-    private String safelyGetOperator()
+    private String safelyGetOperator( boolean isOperatorSubstitutionAllowed )
```

Comment:
Should we also use a more specific name for this flag, as you did below? `isNullValueSubstitutionAllowed`

Reply:
I named both flags different because I wanted to highlight the substitution of an operator and the substitution of the text NV.
From my point of view I think this flag name is fine, but I've been working in this ticket, so maybe it's still fresh in my mind. Do you have a better suggestion for a name?

Reply:
Right, it makes sense Marc.